open-apparel-registry/open-apparel-registry
# Remember Cookie Acceptance, Use "Vietnam"

## Overview

Switches to `localStorage` so we can remember a user's cookie acceptance preference between sessions.

Also switches to using "Vietnam" instead of "Viet Nam".

Connects #324 
Connects #326 

## Demo

Cookie Acceptance variable in `localStorage`:

![image](https://user-images.githubusercontent.com/1430060/54707834-622e2380-4b18-11e9-926e-0ab38b51b24a.png)

"Vietnam" in country dropdown:

![image](https://user-images.githubusercontent.com/1430060/54707856-72de9980-4b18-11e9-8b60-6ee01a6b2267.png)

And in parsed results:

![image](https://user-images.githubusercontent.com/1430060/54707869-7b36d480-4b18-11e9-952c-92fd2dafa27d.png)

## Testing Instructions

* Check out this branch and `server` and `manage migrate`
* Go to [:6543/](http://localhost:6543/)
  - [x] Ensure you see a Cookie Acceptance box
* Accept cookies into your computer
* Once the box is removed, close that tab, open a new tab and go to [:6543/](http://localhost:6543/)
  - [x] Ensure you don't see the Cookie Acceptance box
* Search for "Viet" in the country filter.
  - [x] Ensure you see "Vietnam" as a suggestion
  - [x] Ensure you don't see "Viet Nam" as a suggestion
* Sign up / Log in.
* Upload [15.csv](https://github.com/open-apparel-registry/open-apparel-registry/blob/develop/src/django/api/management/commands/facility_lists/15.csv)
* Run `manage batch_process --action parse --list-id 16` with the correct list id of the newly uploaded list
* Go to that list's detail view
  - [x] Ensure the parsed Vietnamese address says "Vietnam" as the country
